### DISCUSSION TITLE
How to set a TLS certificate for SMTP Server

### DISCUSSION THREAD

**Discussion by drtech981** [postalserver/postal#1829] [Help with using Postal] [answered]
How do I go about setting TLS certificates for the SMTP Server ? I tried using Lets Encrypt issued certificates but that is not working.
Please help !

**Comment by willpower232** [comment]
The docker containers can only see the contents of the config directory so I imagine that may be related. You'll need to ensure the certificate files are copied in there somehow.

**Comment by drtech981** [comment, marked as answer]
Somehow NGINX Proxy Manager issued certificates do not work with Postal SMTP server (RSA vs ECDSA issue). I finally settled for Caddy issued certificates, and it is working fine now.

**Reply by uhlhosting** [reply]
Hi @drtech981 
How did you solved the TLS/SSL issues with Caddy, since it seems now our certificates are not in right format for caddy and we cannot enable SSSL/TLS.

**Reply by The-3x** [reply]
@drtech981 how did you do that?

**Reply by drtech981** [reply]
Add 
```
{                                                                                         
key_type rsa4096                                                              
 }
```       
within the Caddyfile (`/opt/postal/config/Caddyfile`)

**Reply by uhlhosting** [reply]
@drtech981 how you handled the Caddy to release new SSLs since I have already SSL in place, guess removing the ones in place will force Caddy to generate new ones.

**Reply by eZwzAEkx** [reply]
Any how to do this with Caddy?

**Reply by uhlhosting** [reply]
@eZwzAEkx 
just add:

```
{
  key_type rsa4096
}

# Update this to match the hostname you wish to use for accessing the
# Postal web interface.
```
To your Caddyfile .
